experimental/Schemes/SpecialTypes.jl

```diff
@@ -59,6 +59,40 @@ function generic_fraction(a::MPolyQuoLocalizedRingElem, U::PrincipalOpenSubset)
   return lifted_numerator(a)//lifted_denominator(a)
 end
 
+function is_dense(U::PrincipalOpenSubset)
+  return !is_zero_divisor(complement_equation(U))
+end
+
+function is_constant(a::MPolyLocalizedRingElem) 
+  reduce_fraction(a)
+  return is_constant(numerator(a)) && is_constant(denominator(a))
+end
+
+### Already implemented in AA -- but probably buggy?
+#function is_zero_divisor(f::MPolyElem)
+#  iszero(f) && return true
+#  if is_constant(f)
+#    c = coefficients(f)[1]
+#    return is_zero_divisor(c)
+#  end
+#  return !is_zero(quotient(ideal(parent(f), zero(f)), ideal(parent(f), f)))
+#end
+
+function is_zero_divisor(f::MPolyLocalizedRingElem)
+  iszero(f) && return true
+  if is_constant(f)
+    c = coefficients(numerator(f))[1]
+    return is_zero_divisor(c)
```

Comment:
Should also be tested.

Reply:
But I added a test for this. Why wasn't this triggered? I'll look into that.

Reply:
I have not checked by hand, but it was marked automatically. Maybe there was a problem, not sure.